### PR TITLE
Add allowed content types

### PR DESCRIPTION
**Pull request by prodion23**
## Description
Add allowed content types & update template to support built-in types

### CODE REVIEW

File: gen/go/v1/config.pbloader.go [hunk 1]
```diff
@@ -229,4 +228,12 @@ func (x *DataCapture) loadFromEnv(prefix string, defaultValues *DataCapture) {
 			x.BodyMaxProcessingSizeBytes = &wrappers.Int32Value{Value: defaultValues.BodyMaxProcessingSizeBytes.Value}
 		}
 	}
+	if rawVals, ok := getArrayStringEnv(prefix + "ALLOWED_CONTENT_TYPES"); ok {
+		for _, val := range rawVals {
+			x.AllowedContentTypes = append(x.AllowedContentTypes, wrappers.String(val))
+		}
+	} else if len(x.AllowedContentTypes) == 0 && len(defaultValues.AllowedContentTypes) > 0 {
```
Comment (tim-mwangi) [inline gen/go/v1/config.pbloader.go:235]: did we initialize `defaultValues.AllowedContentTypes`?

Comment (prodion23) [reply gen/go/v1/config.pbloader.go:235]: From what I can tell that is defined in the go agent repo? 

https://github.com/hypertrace/goagent/blob/main/config/defaults.go

File: gen/go/v1/config.pbloader.go [hunk 2]
```diff
@@ -39,7 +39,6 @@ func (x *AgentConfig) loadFromEnv(prefix string, defaultValues *AgentConfig) {
 		for _, rawVal := range rawVals {
 			vals = append(vals, PropagationFormat(PropagationFormat_value[rawVal]))
 		}
-		x.PropagationFormats = vals
```
[the remark this answers is not in the document]
Comment (prodion23) [reply gen/go/v1/config.pbloader.go]: Fixed.

File: gen/go/v1/config.pbloader.go (hunk 1 above)
Comment (tim-mwangi) [reply gen/go/v1/config.pbloader.go:236]: ah i see. I see they come in as a parameter into this function.

**Review by tim-mwangi — APPROVED** [review]

**Pull request merged**